### DISCUSSION TITLE
Why do I have two different versions of Xcode Command Line Tools installed?

### DISCUSSION THREAD

**Discussion by NReilingh** [Homebrew/discussions#1411] [Everyday usage]
I'm trying to understand how Homebrew always seems to claim that my Command Line Tools are out of date despite me being very diligent about installing software updates.

I am on macOS 11.2.3 and have Xcode 12.5 installed, and I have opened Xcode to install the updated tools available for that version. Yet Homebrew fails three different checks: "too outdated", "newer release available", and "does not support macOS 11" for Command Line Tools.

Investigating further, I notice that Homebrew seems to check `clang` in /Library/Developer. Running this myself, I see the output:

```
nreilingh@macmini ~ % /Library/Developer/CommandLineTools/usr/bin/clang --version
Apple clang version 11.0.0 (clang-1100.0.33.8)
Target: x86_64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

This is indeed not the correct version for macOS 11. I noticed on [this gist](https://gist.github.com/yamaya/2924292) that there is another directory inside the Xcode.app bundle, and when querying that version of `clang`, I get this output:

```
nreilingh@macmini ~ % /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang --version
Apple clang version 12.0.5 (clang-1205.0.22.9)
Target: x86_64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

This output is identical to what I get when I run `clang -v` from the `/usr/bin` system path.

Based on Homebrew's output, it seems to be recommended to `rm -rf /Library/Developer/CommandLineTools` and reinstall them to resolve (update: I didn't need to reinstall anything -- it just worked after I removed the `/Library/Developer/CommandLineTools` dir), but I want to understand why I am in this state to begin with, considering I haven't done anything out of the ordinary with my Xcode installs. Is the Xcode installer supposed to update the tools in the /Library/Developer path, and the fact that it didn't is a bug? Or should Homebrew not be looking at the `clang` in that location?

**Comment by SMillerDev** [comment]
XCode commandline tools are not the same as general command-line tools. You can update the xcode ones all you want but homebrew depends on the commandline tools package from Apple. If you remove that and run `xcode-select --install` it'll install it's latest version.

**Reply by NReilingh** [reply]
That package is supposed to be updated by Software Update, then? That wasn't happening either. I can see now from `pkgutil` that the `/Library/Developer/CommandLineTools` are primarily installed from the package `com.apple.pkg.CLTools_Executables`, with the SDKs in that folder coming from some additional packages. But interestingly enough, prior to removing and reinstalling with `xcode-select`, _no_ version of these packages were being reported from `pkgutil` despite the files being present there and xcode-select also insisting that they were already installed.

Perhaps now that those pkg receipts are present on my system, the tools will be updated properly by software update in the future.
